### PR TITLE
Backend: One useState for the form

### PR DESCRIPTION
**Pull request by GeorgeHarland**
Also fixed temp_all_contents function. Now properly returns all data from the table

Frontend: https://github.com/kristian-welsh/character-creation-frontend/compare/feat/data-in-one-object?expand=1

### CODE REVIEW

File: docs/dbsetup.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-Creating database and collection:
-------------------------------------
-use app
-db.createCollection("character_sheets")
-
 running the database:
 ---------------------
+(make sure /data folder exists)
```
Comment (kristian-welsh) [inline docs/dbsetup.txt:3]: good doc canges

File: server.py [hunk 2]
```diff
@@ -19,14 +18,14 @@ def endpoint():
     print("Character endpoint reached.")
     print("---------------------------")
 
-    req_dict = request.get_json()
+    req = request.get_json()
 
     try:
-        database = CallDatabase(MONGO_DOCKER_IP, MONGO_DOCKER_PORT, "/app")
+        database = CallDatabase('localhost', MONGO_PORT, "/app")
```
Comment (kristian-welsh) [inline server.py:24]: This makes it work wheen running locally at. the expence of docker changes needed in the future. it's a trade off, probably a good one for right now.

File: server.py [hunk 3]
```diff
@@ -76,10 +75,10 @@ def store_data(self, key, value):
...
         new_cl = self.db["character_sheets"]
         try:
-            results = new_cl.find()
-            print(results)
-            print(type(results))
-            return str(results)
+            results = []
+            for i in new_cl.find():
+                results.append(i)
+            return results
```
Comment (kristian-welsh) [inline server.py:81]: I'm surprised that both of these work, you tested the new one?

**Review by kristian-welsh — COMMENTED** [review]
mostly looks good, small question about the change to collection.find()

File: server.py (hunk 3 above)
Comment (GeorgeHarland) [reply server.py:81]: the deleted one didn't work (maybe after these changes). latter works, frontend logs the object of all the fields in the table

Comment (kristian-welsh) [reply server.py:81]: nice nice. I'll approve, wait for both PRs approvals before merging at the same time

**Review by kristian-welsh — APPROVED** [review]

**Pull request merged**